test/functional/options/defaults_spec.lua

```diff
@@ -328,3 +331,277 @@ describe('XDG-based defaults', function()
     end)
   end)
 end)
+
+
+describe('stdpath()', function()
+  -- Returns an OS friendly version of the unix_path.
+  local function alter_path(unix_path)
+    if helpers.iswin() then
+      return string.gsub(unix_path, '/', '\\')
+    else
+      return unix_path
+    end
+  end
+
+  context('returns a String', function()
+    describe('with "config"' , function ()
+      it('knows XDG_CONFIG_HOME', function()
+        clear({env={
+          XDG_CONFIG_HOME=alter_path('/home/docwhat/.config')
+        }})
+        eq(alter_path('/home/docwhat/.config/nvim'), funcs.stdpath('config'))
+      end)
+
+      it('handles changes during runtime', function()
+        clear({env={
+          XDG_CONFIG_HOME=alter_path('/home/original')
+        }})
+        eq(alter_path('/home/original/nvim'), funcs.stdpath('config'))
+        nvim('command', 'let $XDG_CONFIG_HOME="/home/new"')
```

Comment:
There is `command` function imported, why not use that? I would actually name `nvim` deprecated: in my tests this is normally expressed as `meths.command('…')`, unless there is some function in helpers (`eval`, `command`, etc).